inst/schema/spec.md

```diff
@@ -0,0 +1,237 @@
+    "month": 0,
+    "intervention": "none",
+    "net_use": 0,
+    "irs_use": 0,
+    "prevalence": "low",
+    "resistance": 0.2,
+    "value": 0.4287
+  },
+  {
+    "month": 0,
+    "intervention": "none",
+    "net_use": 0,
+    "irs_use": 0,
+    "prevalence": "low",
+    "resistance": 0.4,
+    "value": 0.4119
+  },
+  {
+    "month": 0,
+    "intervention": "none",
+    "net_use": 0,
+    "irs_use": 0,
+    "prevalence": "low",
+    "resistance": 0.6,
+    "value": 0.6894
+  },
+  {
+    "month": 0,
+    "intervention": "none",
+    "net_use": 0,
+    "irs_use": 0,
+    "prevalence": "low",
+    "resistance": 0.8,
+    "value": 0.6217906241600673
+  },
+  {
+    "month": 0,
+    "intervention": "none",
+    "net_use": 0,
+    "irs_use": 0,
+    "prevalence": "med",
+    "resistance": 0.2,
+    "value": 0.4716
+  },
+  {
+    "month": 0,
+    "intervention": "none",
+    "net_use": 0,
+    "irs_use": 0,
+    "prevalence": "med",
+    "resistance": 0.4,
+    "value": 0.8187
+  },
+  {
+    "month": 0,
+    "intervention": "none",
+    "net_use": 0,
+    "irs_use": 0,
+    "prevalence": "med",
+    "resistance": 0.6,
+    "value": 0.0453
+  },
+  {
+    "month": 0,
+    "intervention": "none",
+    "net_use": 0,
+    "irs_use": 0,
+    "prevalence": "med",
+    "resistance": 0.8,
+    "value": 0.1402
+  },
+  ...
+]
+```
+
+## GET /graph/prevalence/config
+Returns an array of graph configuration object for the prevalence graph.
+
+Properties:
+* a plotly layout object
+* a data object containing all the plotly metadata for the data series as well as the columns defining the series
+
+and optionally:
+* a plotly config object
+
+Schema: [Graphs.schema.json](./Graphs.schema.json)
+
+Plotly documentation: https://plotly.com/javascript/plotly-fundamentals/
+
+### Example 
+```
+{
+    "layout": {
+           "title": "Projected prevalence in under 5 year olds",
+           "xaxis": {
+               "title": 'years of intervention',
+               "showline": true,
+               "tickvals": [12, 24, 36],
+               "ticktext": [1, 2, 3],
+           },
+           "yaxis": {
+               "title": 'prevalence (%)',
+               "showline": true,
+               "range": [0, 100],
+               "autorange": false
+           }
+    },
+    "data": [
+         {
+             x_col: "month",
+             y_col: "value", 
+             id: "none",
+             name: "No intervention",
+             type: "lines",
+             marker: {color: "grey"}
+         },
+         {
+             x_col: "month",
+             y_col: "value", 
+             id: "ITN",
+             name: "Pyrethoid ITN",
+             type: "lines",
+             marker: {color: "blue"}
+         },
+         {
+             x_col: "month",
+             y_col: "value", 
+             id: "PBO",
+             name: "Switch to Pyrethoid-PBO ITN",
+             type: "lines",
+             marker: {color: "aquamarine"}
+         }
+    ],
+    "config": {
```

Comment:
is this one too many levels of `{` ?  This also doesn't look like it conforms to the spec as you're saying an array of graphs is returned?

Reply:
ah yes, this changed, will update